internal/malicious/blocks.go

```diff
@@ -145,6 +147,166 @@ func SendVeryLargeBlockValue(ctx context.Context, original []byte, _ error) (str
+func SendZeroUpkeepID(ctx context.Context, original []byte, _ error) (string, []byte, error) {
+	name := "Send Zero Upkeep ID"
+
+	type modifiedObservation struct {
+		BlockKey          types.BlockKey           `json:"1"`
+		UpkeepIdentifiers []types.UpkeepIdentifier `json:"2"`
+	}
+
+	var ob chain.UpkeepObservation
+	if err := json.Unmarshal(original, &ob); err != nil {
+		return name, nil, err
+	}
+
+	var ids []types.UpkeepIdentifier
+	for i := 0; i < len(ob.UpkeepIdentifiers); i++ {
+		ids = append(ids, types.UpkeepIdentifier("0"))
+	}
+
+	observation := modifiedObservation{
+		BlockKey:          ob.BlockKey,
+		UpkeepIdentifiers: ids,
+	}
+
+	b, err := json.Marshal(observation)
+	return name, b, err
+}
+
+// SendVeryLargeUpkeepIDs produces an encoded json object with very large upkeep IDs
+func SendVeryLargeUpkeepIDs(ctx context.Context, original []byte, _ error) (string, []byte, error) {
+	name := "Very Large Upkeep ID"
+
+	type modifiedObservation struct {
+		BlockKey          types.BlockKey           `json:"1"`
+		UpkeepIdentifiers []types.UpkeepIdentifier `json:"2"`
+	}
+
+	var ob chain.UpkeepObservation
+	if err := json.Unmarshal(original, &ob); err != nil {
+		return name, nil, err
+	}
+
+	var ids []types.UpkeepIdentifier
+	for i := 0; i < len(ob.UpkeepIdentifiers); i++ {
+		keyStr, err := GenerateRandomASCIIString(1000)
+		if err != nil {
+			return name, nil, err
+		}
+
+		ids = append(ids, types.UpkeepIdentifier(keyStr))
+	}
+
+	observation := modifiedObservation{
+		BlockKey:          ob.BlockKey,
+		UpkeepIdentifiers: ids,
+	}
+
+	b, err := json.Marshal(observation)
+	return name, b, err
+}
+
+// SendLeadingZeroUpkeepIDs produces an encoded json object with upkeep IDs with leading zeroes
+func SendLeadingZeroUpkeepIDs(ctx context.Context, original []byte, _ error) (string, []byte, error) {
+	name := "Leading Zero Upkeep IDs With Different Block"
+
+	type modifiedObservation struct {
+		BlockKey          types.BlockKey           `json:"1"`
+		UpkeepIdentifiers []types.UpkeepIdentifier `json:"2"`
+	}
+
+	var ob chain.UpkeepObservation
+	if err := json.Unmarshal(original, &ob); err != nil {
+		return name, nil, err
+	}
+
+	var ids []types.UpkeepIdentifier
+	for i := 0; i < len(ob.UpkeepIdentifiers); i++ {
+		keyStr, err := GenerateRandomASCIIString(1000)
+		if err != nil {
+			return name, nil, err
+		}
+
+		ids = append(ids, types.UpkeepIdentifier(fmt.Sprintf("%s%s", strings.Repeat("0", rand2.Intn(100)), keyStr)))
+	}
+
+	observation := modifiedObservation{
+		BlockKey:          ob.BlockKey,
+		UpkeepIdentifiers: ids,
+	}
+
+	b, err := json.Marshal(observation)
+	return name, b, err
+}
+
+// SendLeadingZeroUpkeepIDsSameBlock produces an encoded json object with upkeep IDs with leading zeroes for the same block key
+func SendLeadingZeroUpkeepIDsSameBlock(ctx context.Context, original []byte, _ error) (string, []byte, error) {
+	name := "Leading Zero Upkeep IDs With Same Block"
+
```

Comment:
I think this test case is probably superfluous now that the block key is separate from upkeep ids. Originally, it was upkeep keys and the idea was to have different block number per key.